core/app/models/spree/payment_method.rb

```diff
@@ -25,6 +25,8 @@ class PaymentMethod < Spree::Base
     include Spree::Preferences::StaticallyConfigurable
 
     def self.providers
+      Spree::Deprecation.warn 'Spree::PaymentMethod.providers will be deleted without replacement. ' \
```

Comment:
It's not without replacement. The replacement is `Rails.application.config.spree.payment_methods`